src/helpers/single-value-providers.ts

```diff
@@ -50,6 +52,25 @@ export class CustomFieldIntegerValueProvider extends SingleValueProvider<number>
     }
 }
 
+export class DaysWithDataValueProvider extends SingleValueProvider<number> {
+    constructor(public dataType: string, public daysInPast: number) {
+        if(daysInPast < 0) {
+            throw new Error('daysInPast must be a non-negative number');
+        }
+        super('days with data');
+    }
+
+    async getValue(): Promise<number | undefined> {
+        const today = startOfDay(new Date());
```

Comment:
This is fine, but just wanted to mention that `startOfToday()` returns the same date as this does.